### ISSUE TITLE
Attempt to export onnx models for supervized case

### ISSUE THREAD

**Issue by mtauraso** [lincc-frameworks/hyrax#538]
When onnx models are exported by the to_onnx verb, the onnx subsystem looks at sample data being passed through the model and prunes the ability to process any unused values from the resultant model. This causes an issue because supervised models use (data, label) as an input, but when a model is inferred, only the data is processed. This generates an onnx model that can only accept (data).

This task is to see if there is a short and correct way to export onnx models which will accept (data, label) in the supervized case. Please do the following steps:

- Make a copy of "test_to_onnx_successful_export" which works on a modified version of the "trained_hyrax" fixture that uses ["image", "label"] as its training data fields.
- This new test should fail specifically at the stage where sample data is passed through the exported model, and the failure mode ought to be as described above
- Then experiment with `torch.jit.script` vs `torch.jit.trace` approach described here: https://stackoverflow.com/questions/76206199/unused-or-variables-only-used-for-control-flow-in-onnx-model to see if it is possible to make the test pass using that method.

If you are unable to make the test pass using only that method, please STOP, commit current progress to your PR, and explain what you believe is going on.
